### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Fridrif27**
[DEMO LINK](https://fridrif27.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Review by AlyonaV22 — APPROVED** [review]
Good Job!

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterTodo, setFilterTodo] = useState('All');
```
Comment (Moroz-Dmytro) [inline src/App.tsx:13]:
```suggestion
  const [filterType, setFilterType] = useState('All');
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
```
Comment (Moroz-Dmytro) [inline src/App.tsx:18]: Would be good to create constants or `enum` for all error messages

File: src/App.tsx (hunk 1 above)
Comment (Moroz-Dmytro) [inline src/App.tsx:13]: Create `enum` for all fitlerTypes

File: src/components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  todos: Todo[];
+  onFilter: React.Dispatch<React.SetStateAction<string>>;
+  filter: string;
+};
+
+export const Footer: React.FC<Props> = ({ todos, onFilter, filter }) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {todos.filter(todo => !todo.completed).length} items left
```
Comment (Moroz-Dmytro) [inline src/components/Footer.tsx:15]: Create variable for `todos.filter(todo => !todo.completed).length`

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        </a>
+
+        <a
+          href="#/completed"
+          className={cn('filter__link', { selected: filter === 'completed' })}
+          data-cy="FilterLinkCompleted"
+          onClick={() => onFilter('completed')}
+        >
+          Completed
+        </a>
```
Comment (Moroz-Dmytro) [inline src/components/Footer.tsx:44]: Use `map` here

**Review by Moroz-Dmytro — APPROVED** [review]
I'll approve it 👍 
But fix these comments in other 2 parts of these task

**Pull request closed without merging**